include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -322,6 +337,92 @@ class argument_parser
+            {
+                if (std::getenv("SEQAN3_NO_VERSION_CHECK") != nullptr) // environment variable was set
+                {
+                    version_check_option = false;
+                    debug_stream << "nop"<< std::endl;
+                }
+                else
+                {
+                    if (detail::is_terminal())
+                    {
+                        std::cerr << R"(
+#######################################################################
+   Automatic Update Notifications
+#######################################################################
+
+ Applications built with SeqAn can check for updates automatically.
+ This ensures you will be informed about new versions and important
+ fixes and is highly recommended. It involves making a web-request
+ to a server and transmitting basic application metadata (of course
+ not your data files!). For privacy implications, please see:
+ https://github.com/seqan/seqan3/wiki/Update-Notifications
+
+    [a] Always perform version checks for this app (the default).
+    [n] Never perform version checks for this app.
+    [y] Yes, perform a version check now, and ask again next time.
+    [s] Skip the version check now, but ask again next time.
+
+ You can manually adjust the behaviour of a single program call by adding option
+      --version-check 1/0 (version check on/off)
+
+ or you can permanently disable the version check by setting the environment
+ variable SEQAN3_NO_VERSION_CHECK.
+
+ Please enter one of [a, n, y, s]
+#######################################################################
+)";
+                        char chr{};
+                        std::cin >> chr;
+
+                        switch (chr)
+                        {
+                            case 'y':
+                            {
+                                version_check_option = true;
+                                break;
+                            }
+                            case 's':
+                            {
+                                version_check_option = false;
+                                break;
+                            }
+                            case 'n':
+                            {
+                                version_check_option = false;
+                                app_version.write_cookie(std::string{"NEVER"});
+                                break;
+                            }
+                            default:
+                            {
+                                version_check_option = true;
+                                app_version.write_cookie(std::string{"ALWAYS"});
+                                break;
+                            }
+                        }
+                    }
+                    else // if !detail::is_terminal()
+                    {
+                        version_check_option = true; // default: check version if you cannot ask the user
```

Comment:
Woah, so I set NEVER in my cookie, and it still performs the check in non-interactive situations? That needs to be changed to check the cookie first. If never, don't do it; if always, don't warn.

Just if it's not set, display the following to std::cerr:
```
#######################################################################
   Automatic Update Notifications
#######################################################################
 This app performs automatic checks for updates, for more information, see:
 https://github.com/seqan/seqan3/wiki/Update-Notifications
#######################################################################
```

Reply:
ops yes you are right. It does not perform the server call but it does show the message every time

Reply:
hm when should this message be printed? Only if always is specified? Any time the check is performed (due to version-check 0 / Always / Default)?